lib/theory/src/Theory/Constraint/Solver/ProofMethod.hs

```diff
@@ -239,8 +239,8 @@ execProofMethod ctxt method sys =
       case method of
         Sorry _                  -> return M.empty
         Solved
-          | null (openGoals sys) -> return M.empty
-          | otherwise            -> Nothing
+          | null (plainOpenGoals sys) -> return M.empty
```

Comment:
Does this change not influence the regressions and Tamarin's behaviour on some examples ?

Reply:
You're right. We have tested that part before, but let's not take any chances. @yavivanov will report back later.

Reply:
Is this change necessary to fix the bug?

Reply:
Yes. The way @yavivanov explained this to me, openGoals does more than just list all open goals and would not be suitable. @yavivanov could you explain in more detail?

Reply:
> Does this change not influence the regressions and Tamarin's behaviour on some examples ?

I ran the regression tests twice and they return OK. I've run them before, but it is important to be sure. 

> Does this change not influence the regressions and Tamarin's behaviour on some examples ?

I ran the regression tests twice for the newest version of the changes (to make sure) and they returned OK, so the change does not seem to influence Tamarin's behavior for these examples.

Reply:
> OK. Is this change necessary to fix the bug?

A bit of a lengthy answer:
The bug had two "manifestations". The first one was fixed through a tweak in the parsing process. The second one was a case, in which Tamarin "thought" that an incorrect proof is correct because the SOLVED keyword was in it. It occurred, when a "by contradiction" was swapped through a "SOLVED". 
openGoals is used when Tamarin does a proof "on its own" and it is used to count how many goals we need to solve until we have a solution and it also annotates the goals with their age and usefulness. 
openGoals "misbehaved" in the case from above, but such cases are not something that the function is made for. It said that we have 0 unsolved goals, when there were actually unsolved goals present (other functions also returned that there were unsolved goals left, but not openGoals). Since the function behaves correctly in the cases, for which it seems to be made for, I inserted a function that simply returns all of the goals, which have their status as "unsolved". The reasons for this change/solution are two: I believe that openGoals does its job correctly and I also want to keep the changes as local possible. 
Another reason is that, upon encountering a "SOLVED", we do not need to do all of the checks and annotations, which openGoals does, we only need to check if there are unsolved goals left.